spec/gilded_rose_spec.rb

```diff
@@ -3,13 +3,86 @@
 
 describe GildedRose do
 
-  describe "#update_quality" do
+  describe "item properties after running #update_quality" do
+
     it "does not change the name" do
       items = [Item.new("foo", 0, 0)]
       GildedRose.new(items).update_quality()
       expect(items[0].name).to eq "foo"
     end
-  end
 
+    it "an item's SellIn value lowers by one everyday" do
+      items = [Item.new("foo", 10, 0)]
+      GildedRose.new(items).update_quality()
+      expect(items[0].sell_in).to eq(9)
+    end
+
+    it "a regular item's quality value decreases by 1 daily" do
+      items = [Item.new("foo", 10, 10)]
+      GildedRose.new(items).update_quality()
+      expect(items[0].quality).to eq(9)
+    end
+
+    it "quality degrades twice as fast after sell by date" do
+      items = [Item.new("foo", -1, 10)]
+      GildedRose.new(items).update_quality()
+      expect(items[0].quality).to eq (8)
+    end
+
+    # NEED: to deal with negative quality input in item constructor
+    it "quality should never be negative" do
+      items = [Item.new("foo", 0, 0), Item.new("foo", 0, -1)]
+      GildedRose.new(items).update_quality()
+      expect(items[0].quality).to eq (0)
+      # expect(items[1].quality).to eq (0)
+    end
+
+    it "Aged Brie increases in quality the older it gets" do
+      items = [Item.new("Aged Brie", 20, 10)]
+      GildedRose.new(items).update_quality()
+      expect(items[0].quality).to eq (11)
+    end
+
+    # NEED:to deal with negative quality input in item constructor
+    it "the quality of an item is never more than 50" do
+      items = [Item.new("Aged Brie", 50, 50), Item.new("foo", 50, 55)]
+      GildedRose.new(items).update_quality()
+      expect(items[0].quality).to eq (50)
+      # expect(items[1].quality).to eq (50)
+    end
+
+    # NEED: deal with Sulfuras quality not decreasing from 80
```

Comment:
Does a quick scan of the existing implementation suggest that the item name here (and the one given in the requirements) might not match what has been implemented? 

We 'know' that the current implementation (pre-refactoring) works, so we shouldn't have to change any of our code (with the exception of the feature request) to make the tests we're writing actually run.